server/auth/auth.go

```diff
@@ -72,12 +72,15 @@ type clientInfo struct {
 	suspended    bool // penalized, disallow new orders
 }
 
-func (client *clientInfo) cancelRatio() float64 {
-	client.mtx.Lock()
-	total, cancels := client.recentOrders.counts()
-	client.mtx.Unlock()
-	// completed = total - cancels
-	return float64(cancels) / float64(total)
+func (auth *AuthManager) checkCancelRatio(client *clientInfo) (cancels, completions int, ratio float64, penalize bool) {
+	var total int
+	total, cancels = client.recentOrders.counts()
+	completions = total - cancels
+	// ratio will be NaN if total is 0, Inf if completed is 0 but cancels > 0.
+	ratio = float64(cancels) / float64(completions)
```

Comment:
I can't find our previous conversations on this, but at the time of spec writing, "completed orders" meant "any order that's no longer active". The idea of a completed order meaning only successfully executed uncancelled orders is not in the spec and wasn't the intention. The ratio was supposed to be `0 < r <  1`, and I think it should stay that way.

Reply:
I'll dig up the convos, but last time we visited this, completed orders meant all swaps for the order finished successfully.  Total is what you are calling completed with that description.

I think the spec is unclear then because when I looked at the code when doing this PR it appeared to violate the spec.  And I was not intending to violate it here.

What I really want though is an intuitive ratio.  

cancels + successes/completions = total

With the above order fates, where the ratio is cancels/total, which puts it on (0,1) and is like a "cancel rate", and cancels/successes which makes it more like a "ratio" of fate types.

The considerations we are facing now is needing a rather large cancel ratio.  I'm perfectly happy with either formula here as long as it's not hard to interpret the ratio.

Reply:
Still trying to find our previous discussion but "An order is considered completed when all matches have fully settled" is the spec wording.  Regardless of wording, I just want to decide on something appropriate and simple and do it.  I'm fine either way.

Reply:
I definitely agree that the spec is unclear. The idea of "completion" meaning success came with your work on the matcher, I think, but cancellation ratio in spec was written before then. 

If you can keep it on (0,1), I'd appreciate it. And I'll update the spec. Do you prefer "cancellation rate" then?

Reply:
Def. need spec clarification.  I don't necessarily need "rate" to be happy about it though.  I'll revise this back to (0,1), but a bit of this PR that's important to retain is the new-user grace period, and that's framed in terms of the ratio interpretation of cancels:successes, so I'll have to tweak that definition as the ratio will no longer be defined appropriately for that.